projects/github-actions/test-results-to-slack/src/index.js

```diff
@@ -0,0 +1,107 @@
+const { setFailed, getInput } = require( '@actions/core' );
+const { context, getOctokit } = require( '@actions/github' );
+const { WebClient, retryPolicies, LogLevel } = require( '@slack/web-api' );
+
+( async function main() {
+	const ghToken = getInput( 'github_token' );
+	if ( ! ghToken ) {
+		setFailed( 'main: Input `github_token` is required' );
+		return;
+	}
+
+	const token = getInput( 'slack_token' );
+	if ( ! token ) {
+		setFailed( 'Input `slack_token` is required' );
+		return;
+	}
+
+	const channel = getInput( 'slack_channel' );
+	if ( ! channel ) {
+		setFailed( 'Input `slack_channel` is required' );
+		return;
+	}
+
+	const username = getInput( 'slack_username' );
+	if ( ! username ) {
+		setFailed( 'Input `slack_username` is required' );
+		return;
+	}
+
+	const icon_emoji = getInput( 'slack_icon_emoji' );
+	if ( ! icon_emoji ) {
+		setFailed( 'Input `slack_icon_emoji` is required' );
+		return;
+	}
+
+	const isFailure = await isWorkflowFailed( ghToken );
+
+	const status = isFailure ? 'failed' : 'passed';
+	let event = context.sha;
+
+	if ( context.eventName === 'pull_request' ) {
+		const { pull_request } = context.payload;
+		event = `PR \`${ pull_request.number }: ${ pull_request.title }\` (${ pull_request.html_url })`;
+	}
+
+	if ( context.eventName === 'push' ) {
+		event = `commit \`${ context.sha } \` on branch \`${ context.ref.substring( 11 ) }\` ${
+			context.payload.head_commit.url
+		}`;
+	}
+
+	const text = `Tests ${ status } for ${ event }`;
```

Comment:
Is this in preparation for future changes?  I assume we didn't really need that extraction just yet, but if you plan on doing more with that in the future (such as not sending any messages for successes), then I guess it makes sense?

```suggestion
	let event = context.sha;

	if ( context.eventName === 'pull_request' ) {
		const { pull_request } = context.payload;
		event = `PR \`${ pull_request.number }: ${ pull_request.title }\` (${ pull_request.html_url })`;
	}

	if ( context.eventName === 'push' ) {
		event = `commit \`${ context.sha } \` on branch \`${ context.ref.substring( 11 ) }\` ${
			context.payload.head_commit.url
		}`;
	}

	const text = `Tests ${ isFailure ? 'failed' : 'passed' } for ${ event }`;
```

Reply:
Yes, this is in preparation for future changes.